### DISCUSSION TITLE
Referential Integrity (RI) Violations That May Not Be there.

### DISCUSSION THREAD

**Discussion by asjones987** [DaxStudio/DaxStudio#629] [Q&A]
Let me start out by saying this may not be a DaxStudio issue, but it is how i first found this issue and hoping someone can provide me some guidance.  I ran DaxStudio 2.15 on my my PBIX Model and found a few RI Violations. Some I knew about and some surprised me.  I looked at the raw data files (Excel) that the model was built off of and things looked good. Everything matched and no blank entries.
For one of the relations I was sure was "good" I put both related fields in a table and everything matched up without extra lines. Next used the DaxStudio Analyze in Excel to look at the related columns. It showed a blank row in the last row of the pivot table when looking at the table. This was confusing as my raw data did NOT have a blank row.
I then jumped into Power BI Desktop and looked at the data view and did not see that blank row there. I then went to Power Query and did not see the blank row (there were not that many rows). Just in case I did filter out blanks.  

After checking again I still have the RI Violation and a blank row in my Analyze in Excel. 

After looking a few of my other tables also have a blank row, but it is NOT showing up in Power BI or my raw data.

Thoughts?

thanks

Alan

**Comment by marcosqlbi** [comment]
You might have values on the "many side" that don't have corresponding values on the "one-side".
See this article for an explanation: 
https://www.sqlbi.com/articles/blank-row-in-dax/

**Reply by asjones987** [reply]
Marco, that is what i first thought something on the "many side", but not on the "one side". However putting each side in at table does not show a blank row and distinct count shows the number I expect.

Power query does not show extra records ... and even to be safe I did have PQ filter out blank records.  However Excel still shows it and RI violation show up

Is it an issue with how Excel and DaxStudio are given the data from Power BI?

![2021-05-26 11_04_31-Organizational - Data Model - Power BI Desktop](https://user-images.githubusercontent.com/30843311/119694069-81b8e900-be12-11eb-8047-cd9c58699aaa.png)

![2021-05-26 11_10_53-Book2 - Excel](https://user-images.githubusercontent.com/30843311/119694921-51be1580-be13-11eb-82ce-97c54e194b33.png)

![image](https://user-images.githubusercontent.com/30843311/119695130-80d48700-be13-11eb-9fc4-3df117f664a3.png)

**Reply by marcosqlbi** [reply]
This is strange. Do you have a repro you can share?

**Reply by asjones987** [reply]
i don't think I can share this... to much org data... what gets me is I just decided to try and but part of the data model from scratch thinking i could remove the internal stuff and show you. However I can't reproduce the issue when starting from a blank PBIX file. I don't know if the bad data is something that was left over from something (like a cache that won't clean). I have gotten in and out of Power BI Desktop, refreshed a zillion times etc.

I really don't want to rebuild this

**Reply by dgosbell** [reply]
I notice in the Power BI screen shots that the many side is left aligned while the one side is right aligned. This indicates that the many side is probably a text column and the one side is numeric. This means that the tabular engine will have to do an implicit cast to resolve the relationship. Its a long shot, but this also means that potentially there could be non-visible characters in the many side values. If the source data is maintained using a web UI I have seen instances where non-breaking spaces can get introduced. 

If you add both columns to the one table visual in Power BI do all the expected values line up or are some missing?

**Reply by asjones987** [reply]
Good catch on the on the text vs numeric. I was looking at that and had just fixed it. I did add both columns to one table and they do line up.   Still stumped on why DaxStudio shows RI and Anaylize in Excel still  show the blank rows. Again don't think it is specifically a DaxStudio issue. 

![2021-05-27 08_15_30-Organizational - Data Model - Power BI Desktop](https://user-images.githubusercontent.com/30843311/119833071-66a6b180-bec4-11eb-8e80-5f7164d58580.png)

**Reply by asjones987** [reply]
Just for documentation wanted to show the raw data 
![2021-05-27 08_23_17-General_Attributes xlsx - Excel](https://user-images.githubusercontent.com/30843311/119837188-f732c100-bec7-11eb-8f14-42523df912ad.png)
![2021-05-27 08_24_36-](https://user-images.githubusercontent.com/30843311/119837193-f863ee00-bec7-11eb-9a5b-90402c31ef58.png)
